daemons/fenced/fenced_history.c

```diff
@@ -363,25 +382,13 @@ stonith_merge_in_history_list(GHashTable *history)
             g_hash_table_lookup(stonith_remote_op_list, op->id);
 
         if (stored_op) {
-            if (stored_op->state != st_failed
-                && stored_op->state != st_done
-                && (op->state == st_failed || op->state == st_done)) {
-                    crm_debug("Updating outdated pending operation %.8s "
-                              "(state=%s) according to the one (state=%s) from "
-                              "remote peer history",
-                              op->id, stonith_op_state_str(stored_op->state),
-                              stonith_op_state_str(op->state));
-
-            } else {
-                continue; // Skip existent
-            }
+            continue; // Skip existent
```

Comment:
This shouldn't happen - right?
So we should be able to skip the lookup and iterate over all entries of the stripped down list we've received.

Reply:
> This shouldn't happen - right?

Not really for now. It was more about "just in case the function is used in other contexts".

> So we should be able to skip the lookup and iterate over all entries of the stripped down list we've received.

Removed this given that this is the only use case so far.